libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Memory/Expressions/Converters/DialogExpressionConverter.cs

```diff
@@ -57,9 +57,14 @@ public override DialogExpression ReadJson(JsonReader reader, Type objectType, Di
                         }
                     }
 #pragma warning disable CA1031 // Do not catch general exception types (return an empty if an exception happens).
-                    catch (Exception)
+                    catch (Exception e)
 #pragma warning restore CA1031 // Do not catch general exception types
                     {
+                        if (e.Message.StartsWith("Cycle detected", StringComparison.InvariantCulture))
```

Comment:
Intead of this approach which is a bit brittle, we could throw a more specific exception in the cycle, and add a new catch on top. For example

```csharp
catch (InvalidOperationException e)
{
    throw;
}
catch (Exception e)
{
    result = new DialogExpression($"='{id}'");
}
```

I'm not sure what the right exception is, I'll put a bit of thought into it, but feel free to propose. Ideally no custom exceptions. Another approach could be fine also, as long as we don't parse the exception text.